src/typechecker.erl

```diff
@@ -3921,6 +3968,15 @@ top() ->
 type_var(Name) ->
     {var, erl_anno:new(0), Name}.
 
+type_record(Name) ->
+    {type, erl_anno:new(0), record, [{atom, erl_anno:new(0), Name}]}.
+
+type_fun(0) ->
```

Comment:
There is no need for this special case. `lists:seq(1,0)` returns `[]`.

Reply:
Fixed